From f3fbc79f66052233ff4c67b17d30c83de40ad574 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Tue, 5 Sep 2017 11:25:47 -0700
Subject: [PATCH] TOOLS-1852 sdc-clients: Fix cnapi test to run even if there
 is a down headnode Reviewed by: Tim Kordas <tim.kordas@joyent.com>

---
 test/cnapi.test.js | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/test/cnapi.test.js b/test/cnapi.test.js
index c7d28c3..151eb1d 100644
--- a/test/cnapi.test.js
+++ b/test/cnapi.test.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 var bunyan = require('bunyan');
@@ -150,7 +150,7 @@ test('cnapi', function (tt) {
                 minTimeout: 1000
             },
             log: new bunyan.createLogger({
-                name: 'cnapi_unit_test',
+                name: 'nodesdcclientstest-cnapi',
                 stream: process.stderr,
                 level: (process.env.LOG_LEVEL || 'info'),
                 serializers: bunyan.stdSerializers
@@ -161,9 +161,13 @@ test('cnapi', function (tt) {
 
     tt.test(' list servers', function (t) {
         cnapi.listServers({ headnode: true }, function (err, servers) {
-            t.ifError(err);
-            t.ok(servers);
-            SERVER = servers[0].uuid;
+            t.ifError(err, err);
+
+            // Choose a running server for subsequent tests.
+            SERVER = servers.filter(function (s) {
+                return (s.status === 'running');
+            })[0].uuid;
+
             t.end();
         });
     });
-- 
2.21.0

